nrf_security/src/core/CMakeLists.txt

```diff
@@ -5,9 +5,27 @@
 #
 
 if (CONFIG_PSA_CORE_BUILTIN)
-  add_subdirectory(builtin)
+  include(builtin/config.cmake)
+elseif(CONFIG_PSA_CORE_OBERON)
+  include(nrf_oberon/config.cmake)
 endif()
 
-if (CONFIG_PSA_CORE_OBERON)
-  add_subdirectory(nrf_oberon)
+if (CONFIG_PSA_CORE_BUILTIN OR CONFIG_PSA_CORE_OBERON)
+  # If either of the above were enabled
+  target_sources(${mbedcrypto_target}
```

Comment:
Shouldn't this be inside COMPILE_PSA_APIS?

Or for simplication: Shouldn't everything inside this file be wrapped in COMPILE_PSA_APIS

Reply:
I don't know, I'm not familiar with that variable.

Perhaps Joakim knows.

Reply:
PSA core is only compiled in secure-only builds and inside the TF-M build. In both cases COMPILE_PSA_APIS is set

When you build "client support" for PSA crypto APIs, the PSA core can't be built. In this case you shouldn't have either builtin or Oberon PSA core

Client support is equivalent with Non-secure (zephyr) image, in this case PSA crypto APIs are provided from TF-M build

Reply:
From what I can tell this is still correct.

The two source files were not guarded by COMPILE_PSA_APIS earlier, and files are still added to the variable src_crypto_core when COMPILE_PSA_APIS  is true.

However I don't think this commit improved a lot, as we intend to remove the builtin fairly quickly, and then we should ideally not split the building of the Oberon PSA core across two files.

Reply:
"However I don't think this commit improved a lot, as we intend to remove the builtin fairly quickly, and then we should ideally not split the building of the Oberon PSA core across two files."

I was not aware we would remove the builtin fairly quickly.

Shall I revert?

Reply:
deleted commit